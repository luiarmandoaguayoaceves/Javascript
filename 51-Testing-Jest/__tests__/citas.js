import Citas from '../js/classes/Citas';

describe('Probar la clase de citas', () => {
   
    const citas = new Citas(); // de manera global para las demas pruebas
    const id = Date.now();

    test('Nueva Cita', () => {
        
        const citaObj = {
            id,
            mascota: 'Kira',
            propietario: 'Luis Armando',
            telefono: '3328358880',
            fecha: '01-12-2023',
            hora:'08:39',
            sintomas: 'Solo duerme'
        }

        citas.agregarCita(citaObj)// usar la funcion a probar

        //Prueba
        expect(citas).toMatchSnapshot();
    })

    test('Actualizar cita', () => {

        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Luis Armando',
            telefono: '3328358880',
            fecha: '01-12-2023',
            hora:'08:39',
            sintomas: 'Solo duerme'
        }

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();

    })

    test('Eliminar citas', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    })
})