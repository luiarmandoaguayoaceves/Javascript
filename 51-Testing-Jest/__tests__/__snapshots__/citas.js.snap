// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "01-12-2023",
      "hora": "08:39",
      "id": 1701571720936,
      "mascota": "Nuevo nombre",
      "propietario": "Luis Armando",
      "sintomas": "Solo duerme",
      "telefono": "3328358880",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar citas 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Probar la clase de citas Nueva Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "01-12-2023",
      "hora": "08:39",
      "id": 1701571720936,
      "mascota": "Kira",
      "propietario": "Luis Armando",
      "sintomas": "Solo duerme",
      "telefono": "3328358880",
    },
  ],
}
`;
